hoomd/test/test_gpu_array.cc

```diff
@@ -410,5 +410,33 @@ UP_TEST(GPUVector_basic_tests)
     UP_ASSERT_EQUAL((unsigned int)vec[7], (unsigned int)678);
     UP_ASSERT_EQUAL((unsigned int)vec[8], (unsigned int)987);
     UP_ASSERT_EQUAL((unsigned int)vec[9], (unsigned int)890);
+
+    // test assignment operator
+    GPUVector<unsigned int> v;
+    v = vec;
+    UP_ASSERT_EQUAL(v.size(), static_cast<unsigned int>(10));
+    UP_ASSERT_EQUAL((unsigned int)v[0], (unsigned int)234);
+    UP_ASSERT_EQUAL((unsigned int)v[1], (unsigned int)123);
+    UP_ASSERT_EQUAL((unsigned int)v[2], (unsigned int)654);
+    UP_ASSERT_EQUAL((unsigned int)v[3], (unsigned int)789);
+    UP_ASSERT_EQUAL((unsigned int)v[4], (unsigned int)321);
+    UP_ASSERT_EQUAL((unsigned int)v[5], (unsigned int)432);
+    UP_ASSERT_EQUAL((unsigned int)v[6], (unsigned int)543);
+    UP_ASSERT_EQUAL((unsigned int)v[7], (unsigned int)678);
+    UP_ASSERT_EQUAL((unsigned int)v[8], (unsigned int)987);
+    UP_ASSERT_EQUAL((unsigned int)v[9], (unsigned int)890);
+    UP_ASSERT(static_cast<bool>(v.getExecutionConfiguration()));
+
+    // test move assignment
```

Comment:
Does this test move assignment? This seems to just test a constructor method.

Don't you want to test `auto new_vec = std::move(old_vect)`?

Reply:
What I believe happens here is the constructor gets called to create a temporary object, then the move assignment operator gets called with the temporary object and the `v` vector as arguments. I haven't stepped over the code to make sure, but based on what I had seen when debugging the GPUVector code earlier in this PR, that is the sequence of operations that are done.

Reply:
The `v = vec` above tests assignment.

The `v = GPUVector<unsigned int>(
        5,
        2,
        std::make_shared<ExecutionConfiguration>(ExecutionConfiguration::GPU));`
test below does not appear to test move assignment as @b-butler indicated. Use 
`v = std::move(vec)` instead to the move constructor.

Reply:
Checking above, move assignment and the move constructor are two different operations. I'm not sure off the top of my head how to trigger move assignment - if possible we should test that as well with a 3rd test.

Reply:
Since `std::move` just marks its argument as an rvalue reference and doesn't choose which of move assignment/move constructor to use, I would think that something like
```
GPUVector<unsigned int> v;
v = std::move(vec);
```
would test move assignment, not move constructor, and something like
```
GPUVector<unsigned int> v = std::move(vec);
```
would test the move constructor.